oauth2/oauth2-resource-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/resourceserver/OAuth2ResourceServerConfigurer.java

```diff
@@ -0,0 +1,140 @@
+package org.springframework.security.config.annotation.web.configurers.oauth2.resourceserver;
+
+import org.springframework.security.config.annotation.web.HttpSecurityBuilder;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.oauth2.jose.jws.JwsAlgorithms;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtVerifier;
+
+import javax.crypto.SecretKey;
+import java.security.PublicKey;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+public class OAuth2ResourceServerConfigurer<B extends HttpSecurityBuilder<B>> extends
+		AbstractHttpConfigurer<JwtVerifierConfigurer<B>, B> {
+
+	public BearerConfigurer bearer() {}
+
+	public class BearerConfigurer {
+		public BearerConfigurer inBody() {}
+
+		public BearerConfigurer inHeader() {}
+
+		public BearerConfigurer inQuery() {}
+
+		public OAuth2ResourceServerConfigurer<B> and() {}
+	}
+
+
+
+	public JwtConfigurer jwt() {}
+
+	public class JwtConfigurer {
+		public JwtDecoderConfigurer decoder() {}
+
+		public JwtConfigurer decoder(JwtDecoder jwtDecoder) {}
+
+		public JwtVerifierConfigurer verifier() {}
+
+		public JwtVerifier verifier(JwtVerifier jwtVerifier) {}
+
+		public OAuth2ResourceServerConfigurer<B> and() {}
+	}
+
+	public class JwtDecoderConfigurer {
+		public JwtDecoderConfigurer rs256() {}
+
+		public JwtDecoderConfigurer algorithm(JwsAlgorithms algorithm) {}
+
+		public JwkSetConfigurer jwkSet() {}
+
+		public JwtConfigurer and() {}
+	}
+
+	public class JwtVerifierConfigurer {
+		public AudiencesConfigurer audiences() {}
+
+		public IssuersConfigurer issuers() {}
+
+		public JwtVerifierConfigurer clockSkew(Integer skew, ChronoUnit unit) {}
+
+		public JwtVerifierConfigurer clockSkew(Duration skew) {}
+
+		public JwtVerifierConfigurer additional(JwtVerifier verifier) {}
+
+		public JwtConfigurer and() {}
+	}
+
+	public class AudiencesConfigurer {
+		public AudiencesConfigurer audience() {}
+
+		public JwtVerifierConfigurer and() {}
+	}
+
+	public class IssuersConfigurer {
+		public IssuersConfigurer issuer() {}
+
+		public JwtVerifierConfigurer and() {}
+	}
+
+	public class JwkSetConfigurer {
+		public RemoteJwkSetConfigurer remote(String location) {}
```

Comment:
JWK Set URL is the only configuration we require as of now. We can provide more configuration later when we implement a new feature, for example, configuration of RSA PK.

Reply:
Okay. Btw, I think I misunderstood your request from earlier, then. Was writing out what I anticipated the configuration would be, including anticipating features we were targeting for this milestone. Happy to remove all of that though (including the opaque token configuration that is also in this PR).

Reply:
Yeah maybe I miscommunicated. We do have to anticipate what the DSL will look like for opaque token and configurable verification key configurations but we won't implement that now.

We just need to ensure the design of the DSL will accommodate this seamlessly when we do implement.